discovery/src/lib.rs

```diff
@@ -129,7 +138,14 @@ impl Builder {
         )?;
 
         #[cfg(not(feature = "with-dns-sd"))]
-        let svc = libmdns::Responder::spawn(&tokio::runtime::Handle::current())?.register(
+        let _svc = if !_bind_ips.is_empty() {
```

Comment:
Is it possible to stick all the `#[cfg(not(feature = "with-dns-sd"))]` stuff in the same block and have the condition just once?
```rust
#[cfg(not(feature = "with-dns-sd"))]
{
    let svc = if !_bind_ips.is_empty() {
        libmdns::Responder::spawn_with_ip_list(&tokio::runtime::Handle::current(), _bind_ips)?
    } else {
        libmdns::Responder::spawn(&tokio::runtime::Handle::current())?
    };
    let svc = svc.register(
        "_spotify-connect._tcp".to_owned(),
        name,
        port,
        &["VERSION=1.0", "CPath=/"],
    );
}
```

Reply:
@setime